application/src/Controller/Admin/ResourceTemplateController.php

```diff
@@ -40,6 +50,271 @@ public function showDetailsAction()
         return $view;
     }
 
+    public function importAction()
+    {
+        $form = $this->getForm(ResourceTemplateImportForm::class);
+        $view = new ViewModel;
+
+        if ($this->getRequest()->isPost()) {
+            $data = $this->params()->fromPost();
+            $form->setData($data);
+            if ($form->isValid()) {
+                $file = $this->params()->fromFiles('file');
+                if ($file) {
+                    // Process import form.
+                    $import = json_decode(file_get_contents($file['tmp_name']), true);
+                    if (JSON_ERROR_NONE === json_last_error()) {
```

Comment:
Probably a little unnecessary complexity here... json_decode will return null on error. Sometimes you need to check for errors explictly but since you already only allow arrays as "valid" data it seems you could collapse this into just checking `importIsValid` and relying on the null being rejected by the check against array there.

This would let you drop an indentation level and a duplicated `else` clause.